experimental/packages/otlp-transformer/src/trace/types.ts

```diff
@@ -30,8 +30,8 @@ export interface IResourceSpans {
   /** ResourceSpans resource */
   resource?: IResource;
 
-  /** ResourceSpans instrumentationLibrarySpans */
-  instrumentationLibrarySpans: IInstrumentationLibrarySpans[];
+  /** ResourceSpans scopeSpans */
+  scopeSpans: IInstrumentationLibrarySpans[];
 
```

Comment:
I added this double-publishing for both Metrics and Traces in [4eae398](https://github.com/open-telemetry/opentelemetry-js/pull/2929/commits/4eae3984b9e8d99953323830fcf2c50545a1f5a2). However, it seems that older collectors do not accept the extra keys `scopeSpans` and `scopeMetrics` being present in the request payload.

Therefore the minimum supported version for the collector using `http/json` still is `0.48` even though old and new keys are present. :slightly_frowning_face:

Reply:
> However, it seems that older collectors do not accept the extra keys `scopeSpans` and `scopeMetrics` being present in the request payload.

Thanks for checking this.
What is the error in this case? Do you know if the current collector >(0.49.0) also behaves this way for new keys which might be added in the future?

Reply:
The error I receive in the collector up to `0.47` is `"unknown field "foo" in v1.ResourceMetrics"` with code `400`.

The current collector (`0.50`) does the same. For instance: adding a field `"foo"` to `resourceMetrics` gives me `unknown field "foo" in v1.ResourceMetrics"` with code `400`

Reply:
Not related to this PR, but if I understand you correctly, it means that every time OTLP proto will add a new field, new JSON exporters will not be compatible with old collectors, right? Unlike the http/proto and grpc variants which should support this pattern.

If this is the case, I believe this behavior might cause problems for us again soon, when we upgrade to a newer proto version and it will require again a new collector. I am not an expert in protobuf, but from what I understood - new fields in proto should just be ignored (fallback to their proto-type default value).

As the one that researched this behavior, do you support creating an issue in the collector repo to discuss the plan for JSON backward compatibility?

Reply:
Yes that is correct as far as I can tell. I'll build a minimal example that can be used with `curl` to demonstrate it and I'm going to create an issue in the collector repo. I'll link the issue in this thread once I'm done :slightly_smiling_face:. 

You're right, It should not be a problem with `grpc` and `proto`, just `json`.

Reply:
I have created the issue on the collector repo: https://github.com/open-telemetry/opentelemetry-collector/issues/5312